app/utilities/decimal_places.py

```diff
@@ -0,0 +1,102 @@
+from decimal import Decimal
+from typing import Literal, TypeAlias
+
+from babel import Locale, numbers, units
+
+UnitLengthType: TypeAlias = Literal["short", "long", "narrow"]
+
+
+def custom_format_decimal(value: int | Decimal | float, locale: Locale | str) -> str:
+    """
+    This function provides a wrapper for the numbers `format_decimal` method, generating the
+    number format (including the desired number of decimals), based on the value entered by the user and
+    the locale.
+    """
+    number_format = get_number_format(value, locale)
+
+    return numbers.format_decimal(
+        value,
+        format=number_format,
+        locale=locale,
+    )
+
+
+def custom_format_currency(
+    value: float | Decimal,
+    currency: str,
+    locale: Locale | str,
+    decimal_limit: int | None = None,
+) -> str:
+    """
+    This function provides a wrapper for the numbers `format_currency` method, generating the
+    number format (including the desired number of decimals).
+
+    The number of decimals displayed is based on the value entered by the user, the decimal limit set in the schema
+    and the locale.
+    """
+    decimal_places = len(str(value).split(".")[1]) if "." in str(value) else 0
```

Comment:
This would mean other decimal symbols like commas won't work. Can you not get the decimal symbol using the locale?

Reply:
Yes it is possible to get the separator with the locale, was doing that previously, but finding it difficult to get this working as the format_decimal method only accepts decimal and float types, so not sure it's possible to pass types to it with other decimal symbols i.e `,` 😕. Would the value already be formatted by the time it gets to these methods?